src/pkg/services/m365/api/drive_pager.go

```diff
@@ -17,6 +17,84 @@ import (
 	"github.com/alcionai/corso/src/pkg/logger"
 )
 
+// ---------------------------------------------------------------------------
+// non-delta item pager
+// ---------------------------------------------------------------------------
+
+var _ itemPager[models.DriveItemable] = &driveItemPageCtrl{}
+
+type driveItemPageCtrl struct {
+	gs      graph.Servicer
+	builder *drives.ItemItemsItemChildrenRequestBuilder
+	options *drives.ItemItemsItemChildrenRequestBuilderGetRequestConfiguration
+}
+
+func (c Drives) NewDriveItemPager(
+	driveID, containerID string,
+	selectProps ...string,
+) itemPager[models.DriveItemable] {
+	options := &drives.ItemItemsItemChildrenRequestBuilderGetRequestConfiguration{
+		QueryParameters: &drives.ItemItemsItemChildrenRequestBuilderGetQueryParameters{
+			Top: ptr.To(maxNonDeltaPageSize),
+		},
+	}
+
+	if len(selectProps) > 0 {
+		options.QueryParameters.Select = selectProps
+	}
+
+	builder := c.Stable.
+		Client().
+		Drives().
+		ByDriveId(driveID).
+		Items().
+		ByDriveItemId(containerID).
+		Children()
+
+	return &driveItemPageCtrl{c.Stable, builder, options}
+}
+
+func (p *driveItemPageCtrl) getPage(ctx context.Context) (PageLinkValuer[models.DriveItemable], error) {
+	page, err := p.builder.Get(ctx, p.options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return EmptyDeltaLinker[models.DriveItemable]{PageLinkValuer: page}, nil
+}
+
+func (p *driveItemPageCtrl) setNext(nextLink string) {
+	p.builder = drives.NewItemItemsItemChildrenRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (c Drives) GetItemsInContainerByCollisionKey(
+	ctx context.Context,
+	driveID, containerID string,
+) (map[string]string, error) {
+	ctx = clues.Add(ctx, "container_id", containerID)
+	pager := c.NewDriveItemPager(driveID, containerID, idAnd("name", "file", "folder")...)
+
+	items, err := enumerateItems(ctx, pager)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "enumerating drive items")
+	}
+
+	m := map[string]string{}
+
+	for _, item := range items {
+		// folders are returned in this query, we only want the files
+		if item.GetFile() == nil {
+			continue
+		}
```

Comment:
Won't we need both files and folders? It's not possible in OneDrive to have a file and folder with the same name. It's a little difficult to manufacture that situation since the web UI doesn't seem to want to let users change the extensions on things but it can be done by uploading a file to OneDrive. Something like
1. make folder foo in root of OneDrive
2. on local machine `touch ~/Downloads/foo`
3. drag & drop new file foo from above to OneDrive root folder
4. OneDrive pops up dialog saying the file has the same name as another item and gives options to replace or keep both

Keeping both results in `foo (1)` being created. Replace appears to result in an error of "something went wrong so we couldn't upload **foo**"

Reply:
Good call, will remove the filter.